### ISSUE TITLE
Already disconnecting

### ISSUE THREAD

**Issue by aikar** [WaterfallMC/Waterfall-Old#44]
Just updated tonight, and getting this occasionally:

```
proxy:01:49:07 [INFO] [/xxxxxxx] <-> InitialHandler has connected
proxy:01:49:08 [SEVERE] Mar 29, 2016 1:49:08 AM io.netty.channel.AbstractChannelHandlerContext invokeExceptionCaught
proxy:WARNING: An exception was thrown by a user handler's exceptionCaught() method while handling the following exception:
proxy:java.lang.IllegalStateException: Already disconnecting
proxy:  at com.google.common.base.Preconditions.checkState(Preconditions.java:173)
proxy:  at net.md_5.bungee.connection.InitialHandler.disconnect(InitialHandler.java:570)
proxy:  at net.md_5.bungee.connection.InitialHandler.disconnect(InitialHandler.java:558)
proxy:  at net.md_5.bungee.connection.InitialHandler$5.done(InitialHandler.java:448)
proxy:  at net.md_5.bungee.connection.InitialHandler$5.done(InitialHandler.java:434)
proxy:  at net.md_5.bungee.http.HttpHandler.done(HttpHandler.java:68)
proxy:  at net.md_5.bungee.http.HttpHandler.channelRead0(HttpHandler.java:59)
proxy:  at net.md_5.bungee.http.HttpHandler.channelRead0(HttpHandler.java:14)
proxy:  at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
proxy:  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:318)
proxy:  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:304)
proxy:  at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:276)
proxy:  at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:263)
proxy:  at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:147)
proxy:  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:318)
proxy:  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:304)
proxy:  at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1110)
proxy:  at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:963)
proxy:  at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:360)
proxy:  at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:244)
proxy:  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:318)
proxy:  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:304)
proxy:  at io.netty.handler.timeout.ReadTimeoutHandler.channelRead(ReadTimeoutHandler.java:152)
proxy:  at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:318)
proxy:  at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:304)
proxy:  at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:846)
proxy:  at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:823)
proxy:  at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:339)
proxy:  at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:255)
proxy:  at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112)
proxy:  at java.lang.Thread.run(Thread.java:745)
```

**Comment by aikar** [comment]
Might be related, but having really weird bugs with command completion disconnecting users or trying to send users to another server just by trying to tab complete.

**Comment by minecrafter** [comment]
That last issue is due to a BungeeCord issue which was fixed upstream (Waterfall has the fix now).

**Comment by aikar** [comment]
w/ latest still has this issue. 
relates to this: https://github.com/WaterfallMC/Waterfall/commit/45e4a61df042ad13cf6580b5e7901e55164c438b

**Comment by aikar** [comment]
my plugin does do this: https://gist.github.com/aikar/5b13189953561b05096445ab4ca5cc0f

But the issue is not happening every server switch. Unsure the trigger yet. I just see it in logs a lot.

**Comment by aikar** [comment]
I think it may relate to timeouts, as I saw a user time out trigger it.

```
smp6:[03:36:49] [Server thread/WARN]: erai1210 moved wrongly!
smp6:[03:37:19] [Server thread/INFO]: erai1210 lost connection: Timed out
smp6:[03:37:19] [Server thread/INFO]: [Empire] erai1210 logged out
smp6:[03:37:26] [Server thread/INFO]: Joshy628 lost connection: Disconnected
smp6:[03:37:32] [Thread-878/INFO]: UUID of player erai1210 is 2d0aad68-688e-4a2c-b703-907625a2f069
smp6:[03:37:32] [Server thread/INFO]: erai1210[/xxx] logged in with entity id 120264 at ([town]-1077.2238749993303, 4.49550403717361, 519.1563116959151)
proxy3:03:37:24 [INFO] [/xx:1585|erai1210] <-> InitialHandler has connected
proxy3:03:37:31 [INFO] [/xx:1593] <-> InitialHandler has connected
proxy3:03:37:32 [INFO] [/xx:1593|erai1210] <-> ServerConnector [smp6] has connected
proxy3:03:37:32 [SEVERE] Mar 30, 2016 3:37:32 AM io.netty.channel.AbstractChannelHandlerContext invokeExceptionCaught
proxy3:WARNING: An exception was thrown by a user handler's exceptionCaught() method while handling the following exception:
proxy3:java.lang.IllegalStateException: Already disconnecting
```

**Comment by Techcable** [comment]
Lets  just ignore the second time someone disconnects.

**Comment by Techcable** [comment]
Fixed by a6d46fb1b4069eab6148df46bfafe110ce7fc580
